### PR TITLE
add bootstrapcrowbar function to set up the postgres db

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
there is still some packaging problem, so we need to wait for https://github.com/crowbar/crowbar-init/pull/17 to gate again

**Comment by MaximilianMeister** [comment]
waiting for https://github.com/crowbar/crowbar-core/pull/576

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -65,6 +66,8 @@ export cinder_netapp_login
 export cinder_netapp_password
 export localreposdir_target
 export want_ipmi=${want_ipmi:-false}
+export want_external_postgres=${want_external_postgres:-false}
+export want_postgres=${want_postgres:-1}
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: do this both variables needs to be documented?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -1747,6 +1750,46 @@ EOF
...
+    # temporarily make it possible to not use postgres until we switched to the new upgrade process
+    # otherwise we would break the upgrade gating
+    [[ $want_postgres = 0 ]] && return
+    if iscloudver 7plus ; then
+        local http_code
+        install_crowbar_init
+        if $want_external_postgres ; then
+            # TODO: setup_external_postgres
+            # curl -X POST --data "username=crowbar&password=crowbar&host=$postgres_ip&port=$postgres_port&database=$postgres_dbname" $crowbar_init_api/database/connect
+            :
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: If this is not implemented, I would sugest to remove from this PR and add in in a different PR (or in a different commit in this PR if is soon), also remove the var

**Comment by aplanas** [comment]
Some notes:
* /postgres/postgresql/
* ~~why not to add this into the prepareinstcrowbar?~~ (reply in IRC)

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -1747,6 +1750,46 @@ EOF
...
+        else
+            # FIXME: when https://github.com/SUSE-Cloud/automation/pull/1105 is merged
+            http_code=`curl -s -X POST --data "username=crowbar&password=crowbar" -o setup-database.txt -w '%{http_code}' $crowbar_init_api/database/new`
+            if ! [[ $http_code =~ [23].. ]] ; then
+                cat setup-database.txt
+                complain 36 "Could not setup PostgreSQL database"
+            fi
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: Not critical, but I see that the POST, the check of the return code, the print of the content and the complain is replicated. Maybe refactor for a call_crowbar_api function?

Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: look at the comment above which refers to https://github.com/SUSE-Cloud/automation/pull/1105
I just wanted to wait for @jdsn coming back to continue on that

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: okay, will remove it for now

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: one of them, yes. thanks

**Comment by aplanas** [comment]
LGTM: +1

**Comment by rsalevsky** [comment]
+1

**Comment by rsalevsky** [comment]
Can you force set the symlink in the non postgresql case to crowbar? I want to enable the linking feature in the package after this PR is merged.

**Comment by MaximilianMeister** [comment]
> Can you force set the symlink in the non postgresql case to crowbar? I want to enable the linking feature in the package after this PR is merged.

@rsalevsky done

**Comment by rsalevsky** [comment]
+1

**Comment by rsalevsky** [comment]
Needs to be adapted after is merged: https://github.com/crowbar/crowbar-init/pull/23

**Pull request merged**